### ISSUE TITLE
Number of iterations

### ISSUE THREAD

**Issue by VietHoang1512** [TorchSSL/TorchSSL#28]
Hi, thank you for your great work.
I'm trying to reproduce the result of VAT on Cifar10-4000 labels. However, I've run the exp for 100.000 iterations but still can not get the reported result of 89.00 (about only 86.00 and it seems to be converged). So can you help me to clarify this?

**Comment by qianlanwyd** [comment]
log for vat cifar10-4000, seed 0
[2021-12-05 06:50:26,826 INFO] 1048000 iteration, USE_EMA: True, {'train/sup_loss': tensor(9.1400e-05, device='cuda:0'), 'train/unsup_loss': tensor(0.1328, device='cuda:0'), 'train/loss_entmin': tensor(0.0440, device='cuda:0'), 'train/total_loss': tensor(0.0426, device='cuda:0'), 'lr': 0.005874884396511934, 'train/prefecth_time': 0.0034196159839630126, 'train/run_time': 0.15932882690429687, 'eval/loss': tensor(0.6446, device='cuda:0'), 'eval/top-1-acc': 0.893, 'eval/top-5-acc': 0.9944}, BEST_EVAL_ACC: 0.8959, at 1018000 iters

Please run 2^20 steps, as the learning rate gets smaller, the acc will increase.

**Comment by qianlanwyd** [comment]
I am trying to submit a paper to ICML and NAACL, so currently I don't have much time to gather all logs.
I will upload all logs after the ICML ddl (1.28)

**Comment by twmht** [comment]
@qianlanwyd 

did you successufuly train VAT with FP16? I found out the gradient is zero when computing adv vector.
